src/Ormolu/Terminal.hs

```diff
@@ -17,114 +17,113 @@ module Ormolu.Terminal
 
     -- * Printing
     put,
-    putS,
-    putSrcSpan,
-    putRealSrcSpan,
+    putShow,
+    putOutputable,
     newline,
   )
 where
 
-import Control.Monad.Reader
+import Control.Monad (forM_)
 import Data.Text (Text)
+import qualified Data.Text as T
 import qualified Data.Text.IO as T
-import GHC.Types.SrcLoc
+import GHC.Utils.Outputable (Outputable)
 import Ormolu.Utils (showOutputable)
 import System.Console.ANSI
-import System.IO (Handle, hFlush, hPutStr)
+import System.IO (Handle, hFlush)
 
 ----------------------------------------------------------------------------
 -- The 'Term' monad
 
--- | Terminal monad.
-newtype Term a = Term (ReaderT RC IO a)
-  deriving (Functor, Applicative, Monad)
+newtype Term = Term [TermOutputNode]
```

Comment:
I'd be surprised if it would make a difference, but as we already depend on `dlist`, what about using them here?

Reply:
I personally dislike `dlist`, preferring `Data.Sequence` more. I do notice that we're only using it in one place; I'll use dlist here, but I'll make a followup PR to replace DList with Seq